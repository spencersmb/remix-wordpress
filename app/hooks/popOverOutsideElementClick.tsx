import type { MutableRefObject } from "react";
import { useEffect } from "react";

/**
 * Hook that alerts clicks outside of the passed ref
 * https://stackoverflow.com/a/42234988/5794430
 */
export function useOutsideAlerter(ref: MutableRefObject<HTMLElement | null>, child: MutableRefObject<HTMLElement | null>, callback: () => void) {
  useEffect(() => {
    /**
     * Alert if clicked on outside of element
     */
    function handleClickOutside(event: any) {

      if ((ref.current && !ref.current.contains(event.target)) && (child.current && !child.current.contains(event.target))) {
        // alert("You clicked outside of me!");
        callback()
      }
    }
    // Bind the event listener
    document.addEventListener("click", handleClickOutside);
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener("click", handleClickOutside);
    };
  }, [callback, child, ref]);
}