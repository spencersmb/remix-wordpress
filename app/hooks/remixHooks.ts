import { useMatches } from "@remix-run/react"
import type { ISelectedMatch } from "~/interfaces/remix"

interface IProps {
  ENV: {
    APP_ROOT_URL: string, PUBLIC_WP_API_URL: string
  }
  menus: IMenu[]
  message: null | string
  metadata: ISiteMetaDataMapped
  user: {
    wpAdmin: boolean, 
    resourceUser: null | IResourceUser
  }
}
export const userStateMatches = () => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const matches = useMatches()
  let selectedMatchUser: undefined | ISelectedMatch = matches.find(match => match.data?.user)
  const state = selectedMatchUser?.data as IProps | undefined
  return {
    state
  }
}

export const metaDataMatches = ():{metadata: ISiteMetaDataMapped} => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const matches = useMatches()
  let selectedMatchMetaData: undefined | ISelectedMatch = matches.find(match => match.data?.metadata)
  if(!selectedMatchMetaData){
    throw new Error('no metadata found')
  }
  const {metadata} = selectedMatchMetaData?.data as IProps
  return {
    metadata
  }
}