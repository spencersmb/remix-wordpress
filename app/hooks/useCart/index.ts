import { useContext, createContext, Dispatch, ReactElement, FunctionComponent } from 'react'
import { IShoppingCartActions, IShoppingCartTypes } from './useCartReducer'
import { IModalTemplate } from '../../components/modals/modalTypes'

interface ICartContextType {
  cart: IShopifyCart,
  dispatch: Dispatch<IShoppingCartActions>
}

export const cartInitialState: IShopifyCart  = {
  checkoutUrl: '',
  id: '',
  isOpen: false,
  lines: []
}
export const CartContext = createContext<ICartContextType>({
  cart: cartInitialState,
  dispatch: () => null
})
CartContext.displayName = 'CartContext'

const useCartContext = () => {
  const context = useContext(CartContext)
  if (!context) {
    throw new Error('useCartContext must be used within a Cart Provider app')
  }
  return context
}

/**
 * @Component useSite
 *
 * Primary context to contain global site data like users, site metadata, menus, etc.
 *
 * Currently used to track logged in Admin user as well as Resource Library user.
 *
 */
const useCart = () => {
  const {cart, dispatch} = useCartContext()

  const openModal = ({template}: {template: FunctionComponent | ReactElement}) => {
    dispatch({
      type: IShoppingCartTypes.MODAL_OPEN,
      payload: {template}
    })
  }

  return {
    openModal,
    cart,
    dispatch
  }
}

export default useCart
