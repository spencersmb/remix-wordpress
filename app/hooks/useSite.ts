import { useContext, createContext} from 'react'

export interface IMenu{
  menuItems: IMenuItem[]
  name: string
  slug: string
}

interface IMenuCourse {
  details: {
    url: string
    name: string
  }
}

export interface IMenuItem {
  featured:{
    courses: IMenuCourse[]
  }
  childItems: {
    edges: {node: IMenuItem}[]
  }
  id: string
  label: string
  path: string
  target: string
  title: string
  parentId: string | null
}

export interface ISiteContextState {
  recentPosts?: IPost[]
  categories?: any[]
  metadata: IMetaData
  menu: IMenu[],
  user: null | {
    isLoggedIn: boolean
  }
}
export const siteInitialState: ISiteContextState  = {
  recentPosts: [],
  categories:[],
  metadata: {
    domain: '',
    description: '',
    language: '',
    siteTitle: '',
    social: {
      youtube: "http://youtube.com/everytues",
      twitter: {
        username: "teelacunningham",
        cardType: "summary",
        url: ''
      },
      pinterest: "http://pinterest.com/teelac",
      instagram: "http://instagram.com/everytuesday",
      facebook: {
        url: "http://facebook.com/everytues",
        defaultImage: {
          altText: "Every-Tuesday Logo Black",
          sourceUrl: "http://etheadless.local/wp-content/uploads/2013/09/et-logo-black.png",
          mediaDetails: {
            "height": 143,
            "width": 510
          }
        }
      }
    },
    title: '',
  },
  menu:[],
  user: null
}
export const SiteContext = createContext<ISiteContextState>(siteInitialState)
SiteContext.displayName = 'SiteContext'

/**
 * useSite
 */

export default function useSite() {
  const site = useContext(SiteContext);
  return site;
}

