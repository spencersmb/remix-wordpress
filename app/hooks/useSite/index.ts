import { useContext, createContext, Dispatch, ReactElement, FunctionComponent } from 'react'
import { ISiteAction, ISiteTypes } from './useSiteReducer'
import { IModalTemplate } from '../../components/modals/modalTypes'
import { siteInfo, socialUrls } from '~/lib/wp/site'

export interface ISiteContextState {
  recentPosts?: IPost[]
  categories?: any[]
  metadata: IMetaData
  menu: IMenu[],
  user: {
    wpAdmin: boolean
    resourceUser: IResourceUser | null
  }
  modal: {
    open: boolean,
    component: IModalTemplate
  },
  commentsModal:{
    show: boolean,
    commentOn: number
    comments: IPostComment[]
    pageInfo: {
      hasNextPage: boolean
      endCursor: string
    }
  }
}
interface ISiteContextType {
  state: ISiteContextState,
  dispatch: Dispatch<ISiteAction>
}

export const siteInitialState: ISiteContextState  = {
  recentPosts: [],
  categories:[],
  metadata: {
    ...siteInfo,
    domain: '',
    social: {
      youtube: socialUrls.youtube,
      twitter: {
        username: "teelacunningham",
        cardType: "summary",
        url: socialUrls.twitter
      },
      pinterest: socialUrls.pinterest,
      instagram: socialUrls.instagram,
      facebook: socialUrls.facebook,
    },
  },
  menu:[],
  user: {
    resourceUser: null,
    wpAdmin: false
  },
  modal:{
    open: false,
    component: null
  },
  commentsModal:{
    show: false,
    commentOn: 0,
    comments: [],
    pageInfo: {
      hasNextPage: false,
      endCursor: ''
    }
  }
}
export const SiteContext = createContext<ISiteContextType>({
  state: siteInitialState,
  dispatch: () => null
})
SiteContext.displayName = 'SiteContext'

const useSiteContext = () => {
  const context = useContext(SiteContext)
  if (!context) {
    throw new Error('useEssGridAuthContext must be used within a Auth Provider app')
  }
  return context
}

/**
 * @Component useSite
 *
 * Primary context to contain global site data like users, site metadata, menus, etc.
 *
 * Currently used to track logged in Admin user as well as Resource Library user.
 *
 */
const useSite = () => {
  const {state, dispatch} = useSiteContext()

  const openModal = ({template}: {template: FunctionComponent | ReactElement}) => {
    dispatch({
      type: ISiteTypes.MODAL_OPEN,
      payload: {template}
    })
  }

  const closeModal = () => {
    dispatch({
      type: ISiteTypes.MODAL_CLOSE,
    })
  }

  const resourecLibraryLogin = (data: {user: IResourceUser}) => {
    dispatch({
      type: ISiteTypes.LOGIN_RESOURCE_USER,
      payload:{
        user: data.user
      }
    })
  }

  const showComments = (data: {
    commentOn: number
    comments: IPostComment[]
    pageInfo:{
      hasNextPage: boolean
      endCursor: string
    }
  }) => {
    dispatch({
      type: ISiteTypes.SHOW_COMMENTS,
      payload: data
    })
  }

  const hideComments = () => {
    dispatch({
      type: ISiteTypes.HIDE_COMMENTS,
    })
  }

  const addComment = (data: {
    comment: IPostComment
  }) => {
    dispatch({
      type: ISiteTypes.ADD_COMMENT,
      payload: data
    })
  }

  const addCommentReply = (data: {
    comment: IPostComment
  }) => {
    dispatch({
      type: ISiteTypes.ADD_COMMENT_REPLY,
      payload: data
    })
  }


  const fetchMoreComments = (data: {comments: IPostComment[], pageInfo: {hasNextPage: boolean, endCursor: string}}) => {
    dispatch({
      type: ISiteTypes.FETCH_MORE_COMMENTS,
      payload: data
    })
  }

  return {
    openModal,
    closeModal,
    resourecLibraryLogin,
    showComments,
    hideComments,
    addComment,
    addCommentReply,
    fetchMoreComments,
    state,
    dispatch
  }
}

export default useSite
