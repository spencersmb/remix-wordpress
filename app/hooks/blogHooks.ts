import { BreakpointEnums } from "@App/enums/breakpointEnums";
import { addClass } from "@App/utils/pageUtils";
import { setWindowUrlParams } from "@App/utils/windowUtils";
import { useLocation } from "@remix-run/react";
import { useCallback, useEffect, useRef } from "react";
import useSite from "./useSite";

export function useSetUrlPageHistory(pageNumber: number) {

  // pageInfo.page
  useEffect(() => {
    if (pageNumber === 1 || !pageNumber) {
      return
    }

    const url = new URL(window.location.href);
    url.searchParams.set('page', pageNumber.toString())
    // window.history.replaceState(`Page: ${pageNumber}`, `Blog: Page ${pageNumber} - Every-Tuesday`, url.href);
    window.history.pushState(`Page: ${pageNumber}`, `Blog: Page ${pageNumber} - Every-Tuesday`, url.href);
    document.title = `Blog: Page ${pageNumber} - Every-Tuesday`

    // if page = 4 - means get the first 40 items
  }, [pageNumber])
}


interface ISetUrlBlogParams {
  category: string,
  pageInfo: {
    page: number,
  }
  categories: ICategoryItem
}
export function useSetUrlBlogParams({
  category,
  pageInfo,
  categories
}: ISetUrlBlogParams) {

  useEffect(() => {

    if (!categories[category]) {
      return
    }

    const setParams = [
      {
        name: 'page',
        value: categories[category].pageInfo.page.toString()
      },
      {
        name: 'cat',
        value: category
      },
    ]
    setWindowUrlParams({
      setParams,
      pageTitle: `Category - ${category} / Page: ${categories[category].pageInfo.page}`,
      tabTitle: 'Blog - Every-Tuesday'
    })
  }, [category, categories, pageInfo])
}

interface IUseFetchCategoryPosts {
  category: string,
  categories: ICategoryItem,
  addCategoryAction: any,
  loadingPosts: any
}

export function useCheckOldBlogPost(post: IPost) {
  const location = useLocation();
  const locationPrevRef = useRef(location.pathname);
  // solve for older blog posts and the iframe issue
  const checkOldIframes = useCallback(() => {

    if (post.tutorialManager.youtube.id) {
      return null
    }

    let blogContent = document.querySelector('.blog-content')
    let pTags = blogContent?.querySelectorAll('p')

    if (!pTags) {
      return null
    }

    pTags.forEach(pTag => {
      const children = Array.from(pTag.children)
      children.find(child => {
        if (child.tagName === 'IFRAME') {
          addClass(pTag, 'embed-responsive')
        }
      })
    })
  }, [post.tutorialManager.youtube.id])

  // on Page Load check for old iframes
  useEffect(() => {

    checkOldIframes()

    return () => {

    }
  }, [checkOldIframes])

  // on location change check for old iframes
  useEffect(() => {

    if (location.pathname !== locationPrevRef.current) {
      // location changed
      checkOldIframes()
    }
    locationPrevRef.current = location.pathname
  }, [checkOldIframes, location.pathname, post.tutorialManager.youtube.id,])

}

export function useHideCommentsOnTransistion() {
  const location = useLocation();
  const locationPrevRef = useRef(location.pathname);
  const { hideComments, state: { commentsModal } } = useSite();
  useEffect(() => {

    if (location.pathname !== locationPrevRef.current && commentsModal.show) {
      // location changed
      hideComments()
    }
    locationPrevRef.current = location.pathname
  }, [commentsModal.show, hideComments, location.pathname])
}

export function useStickySidebarDevices(){

  const { state: { breakpoint } } = useSite();
  const mobilePrevBreakpointRef = useRef(breakpoint);

  useEffect(() => {

    // if the preivous breakpoint was mobile and the current breakpoint is not mobile
    if (mobilePrevBreakpointRef.current !== (BreakpointEnums.desktop || BreakpointEnums.desktopXL) && breakpoint === (BreakpointEnums.desktop || BreakpointEnums.desktopXL)) {
      window.dispatchEvent(new Event('scroll'))
    }

    mobilePrevBreakpointRef.current = breakpoint
  }, [breakpoint])

}