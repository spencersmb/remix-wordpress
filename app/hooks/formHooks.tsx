import SignUpSuccess from "@App/components/modals/signUpSuccess";
import { useEffect } from "react";
import useSite from "./useSite";

interface IuseSignUpEmailSuccessModal {
  type: FetcherTypes
  dataPass: boolean | undefined
}

export function useSignUpEmailSuccessModal({
  type,
  dataPass,
}: IuseSignUpEmailSuccessModal) {

  const { openModal, closeModal } = useSite()
  useEffect(() => {
    if (type === "done" && dataPass) {
      openModal({
        template: <SignUpSuccess
          message='Check your email and click the link inside to complete the signup process!'
          closeModal={closeModal} />
      })
      //@ts-ignore
      // ref.current.reset();
    }
  }, [closeModal, dataPass, openModal, type]);
}

interface IuseSuccessModal {
  status: boolean | undefined
  modalMessage?: string
}
export function useSuccessModal({
  status = false,
  modalMessage,
}: IuseSuccessModal) {

  const { openModal, closeModal } = useSite()
  const defaultMessage = 'Check your email and click the link inside to complete the signup process!'
  useEffect(() => {
    if (status) {
      openModal({
        template: <SignUpSuccess
          message={modalMessage || defaultMessage}
          closeModal={closeModal} />
      })
      //@ts-ignore
      // ref.current.reset();
    }
  }, [closeModal, openModal, status, modalMessage]);
}
interface IuseResetForm {
  status: boolean
  formRef: React.RefObject<any>
}

export function useResetForm({
  status,
  formRef
}: IuseResetForm) {
  useEffect(() => {
    if (status) {
      formRef.current?.reset()
    }
  }, [formRef, status])
}