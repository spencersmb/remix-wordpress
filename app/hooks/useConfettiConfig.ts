import { BreakpointEnums } from "@App/enums/breakpointEnums"
import confetti from "canvas-confetti"
import type { MutableRefObject} from "react";
import { useCallback, useEffect, useState } from "react"
import useSite from "./useSite"

const confettiConfig = {
  mobile: {
    count: 400,
    origin: {
      y: .9,
      x: .5
    },
    scalar: 1,
    spread: {
      a: 50,
      b: 60,
      c: 100
    },
    velocity: {
      a: 55,
      b: 25,
      c: 45
    }
  },
  tablet: {
    count: 400,
    origin: {
      y: .7,
      x: .5
    },
    scalar: 1.7,
    spread: {
      a: 200,
      b: 300,
      c: 100
    },
    velocity: {
      a: 55,
      b: 25,
      c: 45
    }
  },
  desktop: {
    count: 1600,
    origin: {
      y: 1,
      x: .5
    },
    scalar: 1.5,
    spread: {
      a: 150,
      b: 40,
      c: 120
    },
    velocity: {
      a: 125,
      b: 95,
      c: 105
    }
  }
}
export function useConfettiDeviceParams({ref}: {ref: MutableRefObject<HTMLCanvasElement | null>}) {

  const { state: { breakpoint, breakpointLoaded } } = useSite()
  const [state, setState] = useState({
    loaded: false,
    confettiParams: confettiConfig.desktop,
  })

  const confettiTest = useCallback((params: any) => {


    if (!ref.current) {
      return
    }

    var myConfetti = confetti.create(ref.current, { resize: true });
    var count = params.count;
    var defaults = {
      origin: params.origin
    };

    function fire(particleRatio: number, opts: any) {
      myConfetti(Object.assign({}, defaults, opts,
        {
          scalar: params.scalar,
          particleCount: Math.floor(count * particleRatio)
        }));
    }

    fire(0.25, {
      spread: params.spread.a,
      // startVelocity: 55,
      startVelocity: params.velocity.a,
    });
    fire(0.2, {
      spread: params.spread.b,
    });
    fire(0.35, {
      spread: params.spread.c,
      decay: .91,
      scalar: 0.8
    });
    fire(0.1, {
      spread: 120,
      // startVelocity: 25,
      startVelocity: params.velocity.b,
      decay: 0.92,
      scalar: 1.2
    });
    fire(0.1, {
      spread: 120,
      // startVelocity: 45,
      startVelocity: params.velocity.c,
    });
  },[ref])

  useEffect(() => {
    if(!breakpointLoaded){
      return
    }
    if (breakpoint === BreakpointEnums.mobile) {
      setState({
        loaded: true,
        confettiParams: confettiConfig.mobile
      })
    } else if (breakpoint === BreakpointEnums.tablet) {
      setState({
        loaded: true,
        confettiParams: confettiConfig.tablet
      })
    } else {
      setState({
        loaded: true,
        confettiParams: confettiConfig.desktop
      })
    }

  }, [breakpoint, breakpointLoaded])

  useEffect(() => {
    if (breakpointLoaded && state.loaded){
      confettiTest(state.confettiParams)
    }
    // if (breakpointLoaded && !state.loaded) {
    //   console.log('call confetti',state)
    //   confettiTest(state.confettiParams)
    //   setState({
    //     ...state,
    //     loaded: true,
    //   })
    // }

  }, [breakpointLoaded, confettiTest, state])

  return {state}
}