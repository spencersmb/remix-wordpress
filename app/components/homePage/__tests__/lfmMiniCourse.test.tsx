import { siteInitialState } from "@App/hooks/useSite"
import { renderUseSiteProviderUi } from "@TestUtils/providerUtils"
import LfmMiniCourse from "../lfmMiniCourse"
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import UseSiteProvider from "@App/hooks/useSite/useSiteProvider";


describe('LFM MiniCourse Homepage Section', () => {

  it('Should render Snapshot of lfm-mini-course homepage section', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <UseSiteProvider defaultState={siteInitialState}>
            <LfmMiniCourse />
          </UseSiteProvider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Should show View Mini Course Btn when class is closed', () => {
    const { queryByText } = renderUseSiteProviderUi(
      <MemoryRouter>
        <LfmMiniCourse />
      </MemoryRouter>,
      {
        providerProps: siteInitialState
      }
    )

    const btn = queryByText('View Mini Course')
    expect(btn).toBeVisible()
    expect(btn).toHaveAttribute('href', '/learn-font-making')
  })

  it('Should show View Course Btn when class is open w/ External link', () => {
    const { queryByText } = renderUseSiteProviderUi(
      <MemoryRouter>
        <LfmMiniCourse />
      </MemoryRouter>,
      {
        providerProps: {
          ...siteInitialState,
          metadata: {
            ...siteInitialState.metadata,
            courseLaunchBanners: {
              ...siteInitialState.metadata.courseLaunchBanners,
              lfmBanner: {
                ...siteInitialState.metadata.courseLaunchBanners.lfmBanner,
                showBanner: 'true',
              }
            }
          }
        }
      }
    )

    const btn = queryByText('View Course')
    expect(btn).toBeVisible()
    expect(btn).toHaveAttribute('rel', 'noopener noreferrer')
    expect(btn).toHaveAttribute('href', 'https://courses.every-tuesday.com/p/learn-font-making')
  })
})