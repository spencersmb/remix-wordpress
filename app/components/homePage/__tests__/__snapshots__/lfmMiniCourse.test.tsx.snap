// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LFM MiniCourse Homepage Section Should render Snapshot of lfm-mini-course homepage section 1`] = `
<section
  className="relative flex bg-grey-200 tablet:p-0 et-grid-basic"
>
  <div
    className="absolute top-0 left-1/2 -translate-x-1/2 h-[400px] w-[730px] overflow-hidden tablet:h-auto tablet:w-full tablet:left-auto tablet:translate-x-0 tablet:relative tablet:col-start-1 tablet:col-end-full tablet:row-start-1"
  >
    <div
      className="relative block w-full h-full p-0 overflow-hidden"
    >
      <div
        className="relative h-full"
      >
        <video
          autoPlay={true}
          className="opacity-100 transition-opacity duration-600 ease-in-out delay-300"
          data-testid="lazyLoadVideo"
          loop={true}
          muted={true}
          playsInline={true}
        >
          <source
            data-src="https://static.showit.co/file/E0ybNlpeQ8qUB00gPpkPEg/124817/bonnie_b_roll_1_-_web.mp4"
            type="video/mp4"
          />
        </video>
      </div>
    </div>
  </div>
  <div
    className="relative flex flex-col col-span-2 col-start-2 gap-6 p-6 mb-8 text-center bg-white z-2 mt-60 tablet:col-start-4 tablet:col-span-8 tablet:row-start-1 tablet:my-16 tablet:self-center tablet:pb-10 laptop:col-start-8 laptop:col-span-5 max-w-[433px] mx-auto laptop:items-center"
  >
    <div
      className="absolute top-[-43px] left-[45%] -translate-x-1/2 w-[125px]"
    >
      <div
        className="relative flex-1"
        data-testid="imigx-padding-bot"
        style={
          Object {
            "paddingBottom": "78.5997357992074%",
          }
        }
      >
        <div
          className="absolute w-full lazy-load-wrapper lazy-load-wrapper-block lazy-load-image-full transition-all duration-300"
          data-testid="imigx-container"
        >
          <span
            className=" lazy-load-image-background blur"
            style={
              Object {
                "backgroundImage": "url(https://et-website.imgix.net/et-website/images/black-pin-min.png?auto=format&w=20&fit=clip)",
                "backgroundSize": "100% 100%",
                "color": "transparent",
                "display": "inline-block",
                "height": "595px",
                "width": "757px",
              }
            }
          >
            <span
              className=""
              style={
                Object {
                  "display": "inline-block",
                  "height": "595px",
                  "width": "757px",
                }
              }
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="max-w-[270px] mx-auto relative text-5xl font-sentinel__SemiBoldItal text-grey-800 desktop:text-6xl pt-4"
    >
      Free Font Making
      <div
        className="max-w-[150px] mx-auto mt-2"
      >
        <div
          className="relative flex-1"
          data-testid="imigx-padding-bot"
          style={
            Object {
              "paddingBottom": "46.94736842105263%",
            }
          }
        >
          <div
            className="absolute w-full lazy-load-wrapper lazy-load-wrapper-block lazy-load-image-full transition-all duration-300"
            data-testid="imigx-container"
          >
            <span
              className=" lazy-load-image-background blur"
              style={
                Object {
                  "backgroundImage": "url(https://et-website.imgix.net/images/lfm/mini-course/mini-course-title.png?auto=format&w=20&fit=clip)",
                  "backgroundSize": "100% 100%",
                  "color": "transparent",
                  "display": "inline-block",
                  "height": "446px",
                  "width": "950px",
                }
              }
            >
              <span
                className=""
                style={
                  Object {
                    "display": "inline-block",
                    "height": "446px",
                    "width": "950px",
                  }
                }
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <p
      className="text-lg font-semibold leading-7"
    >
      3 part free video series
    </p>
    <p
      data-testid="desc"
    >
      Learn all the basics of what font making is. Find out the tools youâ€™ll need and what sofware is required to get started making fonts in this free 3 part video series.
    </p>
    <a
      className="mt-3 btn btn-primary btn-primary-ring btn-xl"
      href="/learn-font-making"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
    >
      View Mini Course
    </a>
  </div>
</section>
`;
